examples/07_joining_tables_with_FuzzyJoin.py

```diff
@@ -0,0 +1,171 @@
+df['Sum_of_factors'] = df.iloc[:, [5, 6, 7, 8, 9, 10, 11]].sum(axis=1)
+df[['Happiness score', 'Sum_of_factors']].head(3)
+#################################################################
+# Thus, we cannot use them for our prediction model.
+X = df[['Country']]
+y = df[['Happiness score']]
+
+###############################################################################
+# If we want to create a machine learning model which predicts
+# the happiness index of any new country or future date,
+# we will need to include explanatory factors from other tables.
+
+###############################################################################
+# Finding additional tables
+# ---------------------------
+#
+# Let's inspire ourselfes from the factors used by the Happiness report to explain happiness.
+# We will extract data from the World Bank databank using the following function:
+from dirty_cat.datasets import fetch_world_bank_data
+
+#################################################################
+# We then extract GDP per capita by country:
+gdppc = fetch_world_bank_data(data_code='NY.GDP.PCAP.CD', indicator='gdppc').X
+gdppc.head(3)
+
+#################################################################
+# Life expectancy by country:
+life_exp = fetch_world_bank_data('SP.DYN.LE00.IN', 'life_exp').X
+life_exp.head(3)
+
+#################################################################
+# And the legal rights strength by country:
+legal_rights = fetch_world_bank_data('IC.LGL.CRED.XQ', 'legal_rights').X
+legal_rights.head(3)
+
+###############################################################################
+# Joining World Bank tables to our initial one
+# ----------------------------------------------
+#
+# Now, using dirty_cat's :func:`fuzzy_join` function,
+# we need only one line to join two tables
+# without worrying about preprocessing:
+#
+# We add GDP per capita to the initial table:
+from dirty_cat._fuzzy_join import fuzzy_join
+X1 = fuzzy_join(X, gdppc, on=['Country', 'Country Name'])
+X1.head(20)
+#################################################################
+#
+# Now, we see that our :func:`fuzzy_join` succesfully identified the countries,
+# even though some country names differ between tables.
+#
+# For instance, 'Czechia' is well identified as 'Czech Republic' and 'Luxembourg*' as 'Luxembourg'.
+#
+# .. topic:: Note:
+#
+#    This would all be missed out if we were using other methods such as
+#    `pandas.DataFrame.join <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.join.html>`_,
+#    which searches only for exact matches.
+#    In this case, to reach the best result, we would have to manually clean
+#    the data (e.g. remove the * after country name) and look manually
+#    for matching patterns in observations.
+#
+# Dirty_cat's :func:`fuzzy_join` is the perfect function to avoid doing so (and save time) with great results.
+#
```

Comment:
Here we should have a new subsection. Something like "Keeping only the good matches"

Reply:
Thanks, that looks much better now!